keps/sig-api-machinery/4006-transition-spdy-to-websockets/README.md

```diff
@@ -1104,7 +1274,15 @@ For each of them, fill in the following information by copying the below templat
     - Testing: Are there any tests for failure mode? If not, describe why.
 -->
 
-- TBD (complete for beta)
+- Since all failure modes will be visible from a client command, failures should
+  be easy to detect.
+- The most straightforward failures will be error messages returned from a client
+  command.
+- Another possible failure mode would be hanging (or slow responses) from a client
+  command.
```

Comment:
Can we say more about the major kinds of failure modes? E.g. "network is partitioned between X and Y", "network latency between client and apiserver results in ..." ?

Reply:
I've completely updated this `Failure Modes` section. Please let me know what you think.

Reply:
Thanks. The new failure modes cover SPDY->WebSocket migration aspects well.  Let's also include troubleshooting of Websockets since it is new functionality.   If we can leverage knowledge of SPDY failure modes to fill this out, great, otherwise we may need flesh this out for webhook specifically. (See above comment for some ideas of failure modes that came to mind to me,  there are probably others too?)

Reply:
I've updated current failure modes and added two more, including one you suggested. Please let me know what you think.